code/modules/admin/verbs/adminhelp.dm

```diff
@@ -141,3 +141,234 @@ var/list/adminhelp_ignored_words = list("unknown","the","a","an","of","monkey","
+			for(var/i=1, i<surname_found, i++)
+				var/word = ckey(L[i])
+				if(word)
+					forenames[word] = M
+			//ckeys
+			ckeys[M.ckey] = M
+
+	var/ai_found = 0
+	msg = ""
+	var/list/mobs_found = list()
+	for(var/original_word in msglist)
+		var/word = ckey(original_word)
+		if(word)
+			if(!(word in adminhelp_ignored_words))
+				if(word == "ai" && !ai_found)
+					ai_found = 1
+					msg += "<b>[original_word] <A HREF='?_src_=holder;adminchecklaws=\ref[mob]'>(CL)</A></b> "
+					continue
+				else
+					var/mob/found = ckeys[word]
+					if(!found)
+						found = surnames[word]
+						if(!found)
+							found = forenames[word]
+					if(found)
+						if(!(found in mobs_found))
+							mobs_found += found
+							msg += "<b>[original_word] <A HREF='?_src_=holder;adminmoreinfo=\ref[found]'>(?)</A>"
+							if(!ai_found && isAI(found))
+								ai_found = 1
+								msg += " <A HREF='?_src_=holder;adminchecklaws=\ref[mob]'>(CL)</A>"
+							msg += "</b> "
+							continue
+			msg += "[original_word] "
+
+	return msg
+
+/client/verb/adminhelp()
+	set category = "Admin"
+	set name = "Adminhelp"
+
+// Select a category
+	var/msg
+	var/list/type = list ("Gameplay/Job Inquiries", "Rule Issue")
+	var/selected_type = input("Pick a category.", "Admin Help", null, null) as null|anything in type
+	if(selected_type)
+		msg = input("Please enter your message:", "Admin Help", null, null) as text
+
+	//handle muting and automuting
+	if(prefs.muted & MUTE_ADMINHELP)
+		to_chat(src, "<font color='red'>Error: Admin-PM: You cannot send adminhelps (Muted).</font>")
+		return
+
+	adminhelped = 1 //Determines if they get the message to reply by clicking the name.
+
+
+	//clean the input msg
+	if(!msg)
+		return
+	msg = sanitize(msg)
+	if(!msg)
+		return
+	var/original_msg = msg
+
+
+	if(!mob) //this doesn't happen
+		return
+
+	//generate keywords lookup
+	msg = generate_ahelp_key_words(mob, msg)
+
+	// handle ticket
+	var/datum/client_lite/client_lite = client_repository.get_lite_client(src)
+	var/datum/ticket/ticket = get_open_ticket_by_client(client_lite)
+	if(!ticket)
+		ticket = new /datum/ticket(client_lite)
+	else if(ticket.status == TICKET_ASSIGNED)
+		// manually check that the target client exists here as to not spam the usr for each logged out admin on the ticket
+		var/admin_found = 0
+		for(var/datum/client_lite/admin in ticket.assigned_admins)
+			var/client/admin_client = client_by_ckey(admin.ckey)
+			if(admin_client)
+				admin_found = 1
+				src.cmd_admin_pm(admin_client, original_msg, ticket)
+				break
+		if(!admin_found)
+			to_chat(src, "<span class='warning'>Error: Private-Message: Client not found. They may have lost connection, so please be patient!</span>")
+		return
+
+	ticket.msgs += new /datum/ticket_msg(src.ckey, null, original_msg)
+	update_ticket_panels()
+
+
+	//Options bar:  mob, details ( admin = 2, dev = 3, mentor = 4, character name (0 = just ckey, 1 = ckey and character name), link? (0 no don't make it a link, 1 do so),
```

Comment:
Don't include large blocks of commented stuff

Reply:
It's an explanation for it though? Kinda necessary for future edits.

Reply:
underneath theres a big commented block